drivers/watchdog/wdt_nrfx.c

```diff
@@ -21,41 +21,40 @@ struct wdt_nrfx_data {
 
 struct wdt_nrfx_config {
 	nrfx_wdt_t	  wdt;
-	nrfx_wdt_config_t config;
 };
 
 static int wdt_nrf_setup(const struct device *dev, uint8_t options)
 {
-	const struct wdt_nrfx_config *config = dev->config;
-	struct wdt_nrfx_data *data = dev->data;
 	nrfx_err_t err_code;
+	nrfx_wdt_config_t config;
+	const nrfx_wdt_t *wdt = &dev->config.wdt;
```

Comment:
This cannot be compiled successfully as `dev->config` is of the `const void *` type.
Apparently, CI does not try to compile this file for some reason. This should be corrected as well.

Reply:
Thanks for catching it. I used the wrong test (wdt_basic_api) and it didn't catch that silly mistake. Now testing locally by compiling `tests/drivers/watchdog/wdt_basic_reset_none` on `nrf52840dk_nrf52840` and it compiles